src/nwb_conversion_tools/utils/dict.py

```diff
@@ -170,17 +170,31 @@ def dict_deep_update(
     d: dict
         return the updated dictionary
     """
-    if not isinstance(d, collections.abc.Mapping):
+
+    if not isinstance(dict_to_update, collections.abc.Mapping):
         warnings.warn("input to update should be a dict, returning output")
-        return u
+        return dict_with_update_values
+
     if copy:
-        d = deepcopy(d)
-    for k, v in u.items():
-        if isinstance(v, collections.abc.Mapping):
-            d[k] = dict_deep_update(d.get(k, None), v, append_list=append_list, remove_repeats=remove_repeats)
-        elif append_list and isinstance(v, list):
-            for vv in v:
-                d[k] = append_replace_dict_in_list(d.get(k, []), vv, compare_key, list_dict_deep_update, remove_repeats)
+        dict_to_update = deepcopy(dict_to_update)
+
+    for key_to_update, update_values in dict_with_update_values.items():
+        # Update with a dict like object is recursive until an empty dict is found.
+        if isinstance(update_values, collections.abc.Mapping):
+            sub_dict_to_update = dict_to_update.get(key_to_update, dict())
```

Comment:
The thing is though, without a schema available to reference for that data type the value of this key *ought* to be,  just assuming that it's a dictionary may be incorrect. What if the `key_to_update` is `Elecetrodes` within `Ecephys`? Then it should be a list of dicts instead.

Not to mentioned, a `defaultdict(intented_data_type)` would also perhaps be the better approach if you did know what data type it was beforehand.

Reply:
Ahh, but I guess you know from the above `if` checks that the intended update value is either a dict or a list for this case...

Reply:
In the case of  `key_to_update=Electrodes` the values should be a list. These will not enter that specific if condition but it will enter the one just below: